pkg/openapi-gen/openapi-gen.go

```diff
@@ -422,8 +425,37 @@ func buildSchema(deviceEntry *yang.Entry, parentState yang.TriState, parentPath
 					schemaVal.Extensions = make(map[string]interface{})
 				}
 				for _, extension := range dirEntry.Exts {
-					schemaVal.Extensions[fmt.Sprintf("x-%s",
-						extension.Keyword[strings.Index(extension.Keyword, ":")+1:])] = extension.Argument
+					keyword := fmt.Sprintf("x-%s",
+						extension.Keyword[strings.Index(extension.Keyword, ":")+1:])
+					schemaVal.Extensions[keyword] = extension.Argument
+					if keyword == "x-leaf-selection" {
+						// Overrides a leafref if exists
+						*hasLeafref = true
+						extPath := newPathItem(dirEntry, itemPath, parentPath, pathTypeLeafSelection, targetAlias)
+						respGet200 := openapi3.NewResponse()
+						respDesc := "Leaf selection response"
+						respGet200.Description = &respDesc
+
+						leafrefResolverRef := openapi3.NewObjectSchema()
+						leafrefResolverRef.Properties = make(map[string]*openapi3.SchemaRef)
+						leafrefResolverRef.Title = "ref"
+
+						respGet200.Content = openapi3.NewContentWithJSONSchemaRef(&openapi3.SchemaRef{
+							Ref:   fmt.Sprintf("#/components/schemas/%s", LeafRefOptions),
+							Value: leafrefResolverRef,
+						})
+						extPath.Get.AddResponse(200, respGet200)
+						rbRef := &openapi3.RequestBodyRef{
```

Comment:
The get requestBody is the configuration changes, right?

Reply:
yes - the "form context" as I had been calling it

Reply:
If I remember correctly we need to send the Container for the Leaf we're looking values for.
Strictly speaking, even if sending a `body` together with a `GET` that is not common practice, and the HTTP standard doesn't suggest the use of it (here's an interesting discussion about it: https://stackoverflow.com/questions/978061/http-get-with-request-body).

Even though I'm not super concerned about it might make our life harder if we want to make sure the generated `oapi`  spec is valid, as it now fails with:
```
Semantic error at paths./sdn-fabric/v0.1.x/{fabric-id}/switch/{switch-id}/port/{cage-number}/{channel-number}/speed/values.get.requestBody
GET operations cannot have a requestBody.
```

@SeanCondon  Can you reming me why we need the Form Context?

Reply:
actually, I'll take back the validation part, as it's valid for the CLI validator:
```
$ swagger-cli validate openapi.yaml
openapi.yaml is valid
```
it's the online editor at https://editor.swagger.io/ that complains

Reply:
the form context is to send anything else that we have edited in the form over to onos-config - the leaf selection path might refer to some attribute added or edited in it (not in all cases).

we can change this to a PUT or something. I made it as GET as it is technically possible to do, but if it's a problem let's change it.